src/core/server/integration_tests/saved_objects/migrations/group4/v2_md5_to_mv.test.ts

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import type { TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import type { MigrationResult } from '@kbn/core-saved-objects-base-server-internal';
+import { MAIN_SAVED_OBJECT_INDEX } from '@kbn/core-saved-objects-server';
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import {
+  clearLog,
+  deleteSavedObjectIndices,
+  getKibanaMigratorTestKit,
+  readLog,
+  startElasticsearch,
+} from '../kibana_migrator_test_kit';
+import { delay, createType } from '../test_utils';
+import '../jest_matchers';
+
+const logFilePath = Path.join(__dirname, 'v2_md5_to_mv.test.log');
+
+const SOME_TYPE = createType({
+  switchToModelVersionAt: '8.10.0',
+  name: 'some-type',
+  modelVersions: {
+    1: {
+      changes: [],
+    },
+  },
+  mappings: {
+    properties: {
+      field1: { type: 'text' },
+      field2: { type: 'text' },
+    },
+  },
+});
+
+const ANOTHER_TYPE = createType({
+  switchToModelVersionAt: '8.10.0',
+  name: 'another-type',
+  modelVersions: {
+    '1': {
+      changes: [],
+    },
+  },
+  mappings: {
+    properties: {
+      field1: { type: 'integer' },
+      field2: { type: 'integer' },
+    },
+  },
+});
+const ANOTHER_TYPE_UPDATED = createType({
```

Comment:
Could we also add a test with a type that does not have `modelVersions` defined?

Reply:
Added `TYPE_WITHOUT_MODEL_VERSIONS` to the test

Reply:
FWIW this edge case should no longer happen, we have [a wrapper](https://github.com/elastic/kibana/blob/main/packages/core/saved-objects/core-saved-objects-server-internal/src/apply_type_defaults.ts#L16) in place to inject this property for all types.
Thus, any types not defining model versions are considered as `10.0.0`, e.g. [usage-counters](https://github.com/elastic/kibana/blob/main/src/plugins/usage_collection/server/usage_counters/saved_objects.ts#L36,L53).